### PR TITLE
build agent docs

### PR DESCRIPTION
**Pull request by sebader**

### CODE REVIEW

File: src/infra/build-agents/README.md [hunk 1]
```diff
@@ -1,9 +1,81 @@
...
+
 This part of the repository contains the required Bicep template to deploy the infrastructure used for self-hosted Azure DevOps build agent pools, based on Virtual Machine Scale Sets (VMSS).
 
 Furthermore, it deploys a VMSS which is used as Jump Servers to connect to the private resources, such as private AKS clusters, for debugging etc. To connect to the Jump Servers, Azure Bastion is used. This way, even the Jump Servers do not require any public IPs.
 
+To deploy the infrastructure for the self-hosted Agents and all supporting services such as Jump Servers and private DNS zones, a ready-to-use Bicep template plus the corresponding ADO Pipeline is included in this repository. Bicep is being used for this part of the infrastructure instead of Terraform, because using Terraform could create a "chicken-and-egg" problem with the state storage account which requires public access. All further resources are then created using Terraform.
+
+## Infrastructure components
+
+The Bicep template ([`main.bicep`](./main.bicep)) deploys the following infrastructure components:
```
Comment (msimecek) [inline src/infra/build-agents/README.md:19]: I'm missing here some documentation on "how to use" the template - az command and description of the required parameters.

**Review by msimecek — CHANGES_REQUESTED** [review]

File: src/infra/build-agents/README.md (hunk 1 above)
Comment (sebader) [reply src/infra/build-agents/README.md:19]: Should I just link to the pipeline? The pipeline is there for the deployment

Comment (sebader) [reply src/infra/build-agents/README.md]: how about now?

File: docs/reference-implementation/Getting-Started.md [hunk 2]
```diff
@@ -335,6 +335,9 @@ You can now go to the Azure Portal and check the provisioned resources. In the R
 **Stamp Resources**
 ![Azure Stamp Resources](/docs/media/e2e_azure_resources_stamp.png)
 
+For information on how to connect to the data plane, e.g. of AKS follow [this guide](/src/infra/build-agents/README.md#use-jump-server-via-bastion).
+
+
```
Comment (msimecek) [inline docs/reference-implementation/Getting-Started.md:340]:
```suggestion
```

**Review by msimecek — CHANGES_REQUESTED** [review]

**Review by msimecek — APPROVED** [review]

**Pull request merged**